release notes/v0.31.0.md

```diff
@@ -0,0 +1,82 @@
+k6 v0.31.0 is here! :tada: It's a smaller release with some significant performance improvements, a new `http_req_failed` metric and changes to the output subsystem that enable output extensions with [xk6](https://github.com/k6io/xk6)!
+
+## New features
+
+### Output cleanup and registration API ([#1874](https://github.com/loadimpact/k6/pull/1874))
+
+The state of k6's output packages has been a development pain point for a long time which made it difficult to add new outputs in a consistent way. In the refactor done in v0.31.0 this has been mostly addressed and outputs now implement a saner `Output` interface.
+
+In addition to this, an output registry was added to allow external outputs to be easily added. This expands the possibilities of what [xk6](https://github.com/k6io/xk6) can do, so now besides allowing external JS modules, it's also possible to build custom k6 binaries that use external output implementations! This is very useful if you use a system that's currently not supported by [the built-in outputs](https://k6.io/docs/getting-started/results-output), or need some custom handling of the metrics k6 produces.
```

Comment:
```suggestion
In addition to this, it is now possible to implement custom k6 output extensions in Go with [xk6](https://github.com/k6io/xk6)! This is very useful if you use a system that's currently not supported by [the built-in outputs](https://k6.io/docs/getting-started/results-output), or need some custom handling of the metrics k6 produces.

Writing output extensions is done very similarly to [how JS module extensions are currently written](https://k6.io/blog/extending-k6-with-xk6), though instead of calling `js/modules.Register()`, you should implement the new [`Output` interface](https://github.com/loadimpact/k6/blob/2036baeaa83d76ce17aae95fef7e78e3ef24c735/output/types.go#L51-L74) and call [`output.RegisterExtension()`](https://github.com/loadimpact/k6/blob/2036baeaa83d76ce17aae95fef7e78e3ef24c735/output/extensions.go#L47) with your constructor.
```